Load schema when running db:migrate on fresh dbs

mkasztelnik opened the pull request:
### Motivation / Background

Loading database schemas before `db:migrate` for a single database was fixed in #52830. However, this approach fails when multiple databases are defined. This fix addresses the issue by moving schema loading to occur before initiating database migrations.

### Detail

`db:migrate` has different logic when single or multiple databases are defined (https://github.com/rails/rails/blob/main/activerecord/lib/active_record/railties/databases.rake#L92-L102). When multiple DBs are defined, `migration` method is invoked only for the DBs with pending migrations. The fix moves DBs schema load for new DBs before DBs migrations are invoked.

### Additional information

<!-- Provide additional information such as benchmarks, references to other repositories, or alternative solutions. -->

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

zzak commented on activerecord/lib/active_record/tasks/database_tasks.rb:
```diff
@@ -240,14 +240,31 @@ def truncate_all(environment = env)
 
         check_target_version
 
-        initialize_database(migration_connection_pool.db_config)
```
It's possible because this is a public API that someone is depending on this behavior here, maybe we can pass in a kwarg like `skip_initialize`?

mkasztelnik replied:
:+1: `skip_initialize` added - thx for the review.

mkasztelnik committed:
Load schema when running db:migrate on fresh dbs

Loading database schemas before db:migrate for a single database was
fixed in #52830. However, this approach fails when multiple databases
are defined. This fix addresses the issue by moving schema loading to
occur before initiating database migrations.

Fixes #52829

jeremy committed:
Merge branch 'main' into load-schema-on-migrating-multiple-new-databases

jeremy committed:
Drop CHANGELOG entry since this was backported to 8.0
